pkg/kmm/kernelmapping.go

```diff
@@ -21,44 +21,28 @@ type KernelMappingBuilder struct {
 // KernelMappingAdditionalOptions additional options for KernelMapping object.
 type KernelMappingAdditionalOptions func(builder *KernelMappingBuilder) (*KernelMappingBuilder, error)
 
+func NewKernelMappingBuilder() *KernelMappingBuilder {
+	glog.V(100).Infof("Initializing new empty KernelMapping parameter structure")
+
+	return &KernelMappingBuilder{
+		definition: &moduleV1Beta1.KernelMapping{},
+	}
+}
+
 // NewRegExKernelMappingBuilder creates new kernel mapping element based on regex.
 func NewRegExKernelMappingBuilder(regex string) *KernelMappingBuilder {
 	glog.V(100).Infof(
 		"Initializing new regex KernelMapping parameter structure with the following regex param: %s", regex)
 
-	builder := KernelMappingBuilder{
-		definition: &moduleV1Beta1.KernelMapping{
-			Regexp: regex,
-		},
-	}
-
-	if regex == "" {
-		glog.V(100).Infof("The regex of NewRegExKernelMappingBuilder is empty")
-
-		builder.errorMsg = "'regex' parameter can not be empty"
-	}
-
-	return &builder
+	return NewKernelMappingBuilder().WithRegexp(regex)
```

Comment:
Please don't use With* function under New*Builder functions

Reply:
Umm the only other options I can think of are:
1. a helper function that modifies the `kernelMapping.regexp` and then called twice: in `NewRegExKernelMappingBuilder` as well as in `WithRegexp` (to me this is the same think as the current implementation)
2. code duplication in the 2 functions

Please let me know which one you prefer, or if you have a better solution :)